docs/blog/posts/pydantic.md

```diff
@@ -0,0 +1,303 @@
+                                },
+                                "required": ["name", "author"],
+                                "title": "Package",
+                                "type": "object",
+                            }
+                        },
+                        "properties": {
+                            "packages": {
+                                "items": {"$ref": "#/$defs/Package"},
+                                "title": "Packages",
+                                "type": "array",
+                            }
+                        },
+                        "required": ["packages"],
+                        "title": "Packages",
+                        "type": "object",
+                    },
+                },
+            }
+        ],
+        tool_choice={
+            "type": "function",
+            "function": {"name": "Requirements"},
+        },
+    )
+
+    resp = json.loads(resp.choices[0].message.tool_calls[0].function.arguments)
+    ```
+
+Now, notice in this example that the prompts we use contain purely the data we want, where the tools and tools choice now capture the schemas we want to output. This speration of concerns makes it much easier organize the 'data' and the 'description' of the data that we want back out.
+
+```python
+from typing import List
+from pydantic import BaseModel
+from openai import OpenAI
+
+client = OpenAI()
+
+
+class Package(BaseModel):
+    name: str
+    author: str
+
+
+class Packages(BaseModel):
+    packages: List[Package]
+
+
+resp = client.chat.completions.create(
+    model="gpt-3.5-turbo",
+    messages=[
+        {
+            "role": "user",
+            "content": "Pydantic and FastAPI?",
+        },
+    ],
+    tools=[
+        {
+            "type": "function",
+            "function": {
+                "name": "Requirements",
+                "description": "A list of packages and their first authors.",
+                "parameters": Packages.model_json_schema(),
+            },
+        }
+    ],
+    tool_choice={
+        "type": "function",
+        "function": {"name": "Requirements"},
+    },
+)
+
+Packages.model_validate_json(
+    resp.choices[0].message.tool_calls[0].function.arguments
+)
+```
+
+```json
+{
+  "packages": [
+    {
+      "name": "pydantic",
+      "author": "Samuel Colvin"
+    },
+    {
+      "name": "fastapi",
+      "author": "Sebastián Ramírez"
+    }
+  ]
+}
+```
+
+## Using `pip install instructor`
+
+Although this example may seem contrived, it demonstrates how Pydantic can be used to generate structured data from language models. To simplify this pattern, I have developed a small library called `instructor` that patches the `OpenAI` client. This library offers convenient features such as JSON mode, function calling, tool usage, and open source models.
+
+To achieve the same result using `instructor`, simply follow these steps:
```

Comment:
I think it would be just a bit punchier if it called out the specific changes for `instructor` (and improvements it provides) prior to showing the self-contained code example which I think makes it a bit slower to process. E.g.:
```suggestion
To use `instructor` in the example above, you'd just change `client = OpenAI()` to `client = instructor.patch(OpenAI())`, and then change the client request to:
\`\`\`python
packages = client.chat.completions.create(
    model="gpt-3.5-turbo",
    messages=[
        {
            "role": "user",
            "content": "Pydantic and FastAPI?",
        },
    ],
    response_model=Packages,
)
\`\`\`

In context:
```

(You'd need to change the \\\` to ` in the snippet above, that just broke the github rendering of it, not sure if there's a better way to try to include a code block in a suggestion lol.)